crates/pallet-domains/src/domain_registry.rs

```diff
@@ -63,13 +68,13 @@ impl DomainConfig {
 }
 
 #[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq)]
-pub struct DomainObject<Number, Hash, AccountId> {
+pub struct DomainObject<Number, AccountId> {
     /// The address of the domain creator, used to validate updating the domain config.
     pub owner_account_id: AccountId,
     /// The consensus chain block number when the domain first instantiated.
     pub created_at: Number,
     /// The hash of the genesis execution receipt for this domain.
-    pub genesis_receipt_hash: Hash,
+    pub genesis_receipt_hash: H256,
```

Comment:
why a hard type instead ?

Reply:
`genesis_receipt_hash` is returned from the host function, which can't be generic type.